patches/server/0752-Fix-endermen-shulkers-tp-ing-above-logical-height.patch

```diff
@@ -0,0 +1,81 @@
+-                        this.level.getCraftServer().getPluginManager().callEvent(teleport);
+-                        if (!teleport.isCancelled()) {
+-                            Location to = teleport.getTo();
+-                            blockposition1 = new BlockPos(to.getX(), to.getY(), to.getZ());
+-                        } else {
+-                            return false;
++                    // Paper start - check if the place to teleport is above the dimension's logical height
++                    if (blockposition1.getY() < this.level.dimensionType().logicalHeight()) {
++                        Direction enumdirection = this.findAttachableSurface(blockposition1);
++
++                        if (enumdirection != null) {
++                            // CraftBukkit start
++                            EntityTeleportEvent teleport = new EntityTeleportEvent(this.getBukkitEntity(), this.getBukkitEntity().getLocation(), new Location(this.level.getWorld(), blockposition1.getX(), blockposition1.getY(), blockposition1.getZ()));
++                            this.level.getCraftServer().getPluginManager().callEvent(teleport);
++                            if (!teleport.isCancelled()) {
++                                Location to = teleport.getTo();
++                                blockposition1 = new BlockPos(to.getX(), to.getY(), to.getZ());
++                            } else {
++                                return false;
++                            }
++                            // CraftBukkit end
++                            this.unRide();
++                            this.setAttachFace(enumdirection);
++                            this.playSound(SoundEvents.SHULKER_TELEPORT, 1.0F, 1.0F);
++                            this.setPos((double) blockposition1.getX() + 0.5D, (double) blockposition1.getY(), (double) blockposition1.getZ() + 0.5D);
++                            this.entityData.set(Shulker.DATA_PEEK_ID, (byte) 0);
++                            this.setTarget((LivingEntity) null);
++                            return true;
+                         }
+-                        // CraftBukkit end
+-                        this.unRide();
+-                        this.setAttachFace(enumdirection);
+-                        this.playSound(SoundEvents.SHULKER_TELEPORT, 1.0F, 1.0F);
+-                        this.setPos((double) blockposition1.getX() + 0.5D, (double) blockposition1.getY(), (double) blockposition1.getZ() + 0.5D);
+-                        this.entityData.set(Shulker.DATA_PEEK_ID, (byte) 0);
+-                        this.setTarget((LivingEntity) null);
+-                        return true;
+                     }
++                    // Paper end
+                 }
+             }
+ 
```

Comment:
Idk if I'm reading this right, but this block of diff is WAY bigger than it has to be. Don't change indentation if you don't need to, if you need another conditional fine, but dont indent everything in it to keep the diff down to a minimum.

Reply:
Done.